### PR TITLE
Add Number entity type, and update README docs for Switch and Button

### PR DESCRIPTION
**Pull request by DavidMikeSimon**
* Add the `Number` entity, which can control a single numeric value (just like how `Text` can control a single string)
  * Add docs for `Number` to the `README`
  * Add basic tests for `Number` (similar to the `Text` tests)
* Update some existing docs in `README`:
  * Add documentation for `Button`, which was undocumented before.
    * Document that you have to call `write_config()` to make buttons discoverable
  * Add example code to handle the typical payloads for a `Switch`
  * Update examples to make it clearer that, when you receive a command from HA, you also have to update the MQTT state on your entity (e.g. when you get an `"ON"` payload for a `Switch`, it's your job to call `on()`)

# License Acceptance

- [x] This repository is Apache version 2.0 licensed and by making this PR, I am contributing my changes to the repository under the terms of the Apache 2 license.

# Type of changes

- [ ] Add/update a helper script
- [ ] Add/update link to an external resource like a blog post or video
- [ ] Bug fix
- [x] New feature
- [ ] Test updates
- [ ] Text cleanups/updates

# Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/unixorn/ha-mqtt-discovery/blob/main/Contributing.md) document.
- [x] All new and existing tests pass.
- [x] Any scripts added use `#!/usr/bin/env interpreter` instead of potentially platform-specific direct paths (`#!/bin/sh` is an allowed exception)
- [x] Scripts added/updated in this PR are all marked executable.
- [x] Scripts added/updated in this PR _do not_ have a language file extension unless they are meant to be sourced and not run standalone. No one should have to know if a script was written in bash, python, ruby or whatever. Not including file extensions makes it easier to rewrite the script in another language later without having to change every reference to the previous version.
- [x] I have confirmed that any links added or updated in my PR are valid.

### CODE REVIEW

**Comment by DavidMikeSimon** [comment]
Heya, I'm noticing that the linter is unhappy about `sensors.py`, but most of the lines it's complaining about were not changed in my PR. Should I just reformat the whole file to make it pass the line length lint?

File: README.md [hunk 1]
```diff
@@ -145,19 +185,60 @@ settings = Settings(mqtt=mqtt_settings, entity=switch_info)
...
+from ha_mqtt_discoverable.sensors import Number, NumberInfo
+from paho.mqtt.client import Client, MQTTMessage
+
+# Configure the required parameters for the MQTT broker
+mqtt_settings = Settings.MQTT(host="localhost")
+
+# Information about the `number` entity.
+number_info = NumberInfo(name="test", min=0, max=50, mode="slider", step=5)
+
+settings = Settings(mqtt=mqtt_settings, entity=switch_info)
```
Comment (philharlow) [inline README.md:221]: Vestigial `switch_info`
`settings = Settings(mqtt=mqtt_settings, entity=number_info)`

Comment (DavidMikeSimon) [reply README.md]: @philharlow Updated, along with some similar pre-existing mis-names.

**Comment by DavidMikeSimon** [comment]
@unixorn Bump. Anything you'd like me to do on this PR? I can reformat that file to pass the linter, though I'm not sure about how it was passing the linter in the first place.

**Comment by unixorn** [comment]
@DavidMikeSimon, if you're willing, I'd appreciate the formatting help.

Sorry I've been slow, work's been busy busy.

**Comment by DavidMikeSimon** [comment]
@unixorn Ok, lints passing now!

**Review by unixorn — APPROVED** [review]
Thanks for the help!

**Pull request merged**